src/cloud/digitalocean/provisioner.ts

```diff
@@ -99,26 +101,48 @@ class Provisioner {
 
   /**
    * One-click destruction of a VM
-   * See freedom-module.json for return and error types
-   * @todo currently doesnt wait for destroy to complete before resolving
    * @param {String} name of VM to create
-   * @return {Promise.<Object>}
+   * @return {Promise.<void>}
    */
-  public stop = (name: string) : Promise<Object> => {
-    return this.doRequest_('GET', 'droplets').then((resp: any) => {
-      for (var i = 0; i < resp.droplets.length; i++) {
-        if (resp.droplets[i].name === name) {
-          return Promise.resolve({
-            droplet: resp.droplets[i]
-          });
+  public stop = (name: string): Promise<void> => {
+    this.sendStatus_('STOP');
+    return this.doOAuth_().then((oauthObj: any) => {
+      this.state_.oauth = oauthObj;
+    }).then(() => {
+      return this.destroyServer_(name);
+    });
+  }
+
+  /**
+   * Destroys cloud server; assumes OAuth has already been completed
+   * @todo DELETE request does not return a response body so we
+   * need to check that the response header indicates success (204)
+   * @param {String} droplet name, as a string
+   * @return {Promise.<void>}
+   */
+  private destroyServer_ = (name: string): Promise<void> => {
+    if (this.state_.oauth) {
```

Comment:
Rather than wrapping the whole function in a try/catch block you could make this a precondition-like block:
```typescript
if (!this.state_.oauth) {
  return Promise.reject({
    'errcode': 'OAUTH_ERR',
    'message': 'Cannot destroy server - not logged into digital ocean'
  });
  ...
  ...
}
```

Actually...since this is a private method only called by `stop`, do you really need this check?